RHDEVDOCS 4233 add information on Pipelines authentication using workspaces, container registries (#70384)
openshift/openshift-docs: modules/op-using-ssh-authentication-in-git-type-tasks.adoc

// This module is included in the following assemblies:
// * secure/authenticating-pipelines-repos-using-secrets.adoc

[id="op-using-ssh-authentication-in-git-type-tasks_{context}"]
= Using Git SSH authentication in tasks

> You can invoke directly Git commands in the steps of a task and use SSH authentication, but an additional step is required.

Reviewer: Tiny adverb position and [active voice](https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#voice) nit:
```suggestion
You can directly invoke Git commands in the steps of a task and use SSH authentication, but you must perform an additional step.
```

{pipelines-shortname} provides the SSH files in the `/tekton/home/.ssh` directory and sets the `$HOME` variable to `/tekton/home`. However, Git SSH authentication ignores the `$HOME` variable and uses the home directory specified in the `/etc/passwd` file for the user. Therefore, a step that uses Git command must symlink the `/tekton/home/.ssh` directory to the home directory of the associated user.

For example, if the task runs as the `root` user, the step must include the following command before Git commands:

[source,yaml]
----
apiVersion: tekton.dev/v1
kind: Task
metadata:
  name: example-git-task
spec:
  steps:
    - name: example-git-step
#     ...
      script:

>         ln -s tekton/home/.ssh /root/.ssh

Reviewer 2: `ln -s /tekton/home/.ssh …` or better `ln -s $HOME/.ssh …`.

#     ...
----

However, explicit symlinks are not necessary when you use a pipeline resource of the `git` type or the `git-clone` task available in the Tekton catalog.

> As an example of using SSH authentication in `git` type tasks, refer to link:https://github.com/tektoncd/pipeline/blob/main/examples/v1/taskruns/authenticating-git-commands.yaml[authenticating-git-commands.yaml].

Reviewer: This is even further out of scope for this review, so I'm only pointing it out this time, but this is another instance of a problematic link to a GitHub-hosted repository.

Author: Again, I believe the repo is maintained by Red Hat engineers. There are many links to this and other similar repos in this doc set and removing them would be a major problem. As far as I am aware, the intend of the guidance is to disallow repos that we don't control and that therefore can be volatile.

Reviewer 3: The examples could still change out from under us and not necessarily have been QE verified, particularly for the docs version they might be appearing in (especially if just targeting the `main` branch like this example is).

Author: I have requested Pipelines QE feedback on these examples.
